pandas/core/series.py

```diff
@@ -623,6 +623,29 @@ def view(self, dtype=None):
         return self._constructor(self._values.view(dtype),
                                  index=self.index).__finalize__(self)
 
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        inputs = tuple(
+            x._values if isinstance(x, type(self)) else x
+            for x in inputs
+        )
+        # for binary ops, use our custom dunder methods
+        result = ops.maybe_dispatch_ufunc_to_dunder_op(
+            self, ufunc, method, *inputs, **kwargs)
+        if result is not None:
+            return result
+
+        if hasattr(self._values, '__array_ufunc__'):
+            result = self._values.__array_ufunc__(
+                ufunc, method, *inputs, **kwargs)
+        else:
+            result = np.array(self._values).__array_ufunc__(
+                ufunc, method, *inputs, **kwargs)
+        if result is NotImplemented:
+            raise TypeError("The '{0}' operation is not supported for "
+                            "dtype {1}.".format(ufunc.__name__, self.dtype))
+        return self._constructor(result, index=self.index,
```

Comment:
Note that not every ufunc returns one result, so you should either handle or explicitly error for the other cases. See NDArrayOperatorsMixin for how to handle this.

Reply:
Out of curiosity, do you have the same problem with the current implementation of `__array_wrap__` ?

Reply:
To answer my own question: yes, at least for ufuncs that return a scalar:

```
In [33]: s = pd.Series([1, 2, 3])

In [34]: np.add.reduce(s)
Out[34]: 
0    6
1    6
2    6
dtype: int64

In [35]: pd.__version__
Out[35]: '0.23.4'
```

But for ufuncs that return multiple values it seems to work with pandas master (wrapping each return element I suppose).